kymatio/scattering2d/frontend/__init__.py

```diff
@@ -5,22 +5,25 @@
 
 class Scattering2D(object):
     def __init__(self, *args, **kwargs):
+        frontend_suffixes = {'torch' : 'Torch', 'numpy' : 'NumPy', 'tensorflow'
+                : 'TensorFlow'}
         if 'frontend' not in kwargs:
             warnings.warn("Torch frontend is currently the default, but NumPy will become the default in the next"
                           " version.", PendingDeprecationWarning)
             frontend = 'torch'
         else:
-            frontend = kwargs['frontend']
+            frontend = kwargs['frontend'].lower()
             kwargs.pop('frontend')
 
         try:
             module = importlib.__import__(frontend + '_frontend', globals(), locals(), [], 1)
 
             # Create frontend-specific class name by inserting frontend name
             # after `Scattering`.
+            frontend = frontend_suffixes[frontend]
+
             class_name = self.__class__.__name__
-            class_name = (class_name[:-2] + frontend.capitalize()
-                          + class_name[-2:])
+            class_name = (class_name[:-2] + frontend + class_name[-2:])
```

Comment:
why is it modified here?

Reply:
We use a dictionary to get the frontend name so that we can import the correct class.

Reply:
is it realted to tf and couldn't it be done elsewhere?

Reply:
Not sure. It should have been modified in the numpy 2d pr.

Reply:
so let's remove it, and let's add it in a separated PR

Reply:
Since NP wasn't merged at the time this PR was started, it appears as a diff here. What we have to do is rebase the branch on top of the updated `kymatio-v2` and then these wrinkles should shake out. I suggest we wait until #467 is merged, however, since that will also affect the work here.